### ISSUE TITLE
Object reference not set to an instance of an object.

### ISSUE THREAD

**Issue by phwoomp** [geluk/pass-winmenu#73]
Hello, 

First of all: I Love this program, it's really good! Keep up the good work!

I have been getting some errors recently when using different parts of the program that all say:

**Object reference not set to an instance of an object.**

When I use _Add Password_ from the menu it will add the new password to the local repo just fine but I get this .NET error after:

> ************** Exception Text **************
> System.NullReferenceException: Object reference not set to an instance of an object.
>    at PassWinmenu.ExternalPrograms.Git.CommitFile(String filePath)
>    at PassWinmenu.Windows.DialogCreator.AddPassword()
>    at PassWinmenu.WinApi.Notifications.<>c__DisplayClass11_0.<AddMenuActions>b__1(Object sender, EventArgs args)
>    at System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
>    at System.Windows.Forms.ToolStripMenuItem.OnClick(EventArgs e)
>    at System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
>    at System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
>    at System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
>    at System.Windows.Forms.ToolStripDropDown.OnMouseUp(MouseEventArgs mea)
>    at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
>    at System.Windows.Forms.Control.WndProc(Message& m)
>    at System.Windows.Forms.ToolStrip.WndProc(Message& m)
>    at System.Windows.Forms.ToolStripDropDown.WndProc(Message& m)
>    at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
> 
> 
> ************** Loaded Assemblies **************
> mscorlib
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4250.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
>  
> pass-winmenu
>     Assembly Version: 1.11.0.0
>     Win32 Version: 1.11
>     CodeBase: file:///C:/Users/WINUSER/pass-winmenu/pass-winmenu.exe
>  
> PresentationFramework
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4270.0
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework/v4.0_4.0.0.0__31bf3856ad364e35/PresentationFramework.dll
>  
> WindowsBase
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4270.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/WindowsBase/v4.0_4.0.0.0__31bf3856ad364e35/WindowsBase.dll
>  
> System.Core
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4270.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
>  
> System
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4200.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
>  
> PresentationCore
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4270.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/PresentationCore/v4.0_4.0.0.0__31bf3856ad364e35/PresentationCore.dll
>  
> System.Xaml
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4270.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xaml/v4.0_4.0.0.0__b77a5c561934e089/System.Xaml.dll
>  
> System.Configuration
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4190.0 built by: NET48REL1LAST_B
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
>  
> System.Xml
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4084.0 built by: NET48REL1
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
>  
> PresentationFramework.Aero2
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4084.0 built by: NET48REL1
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework.Aero2/v4.0_4.0.0.0__31bf3856ad364e35/PresentationFramework.Aero2.dll
>  
> Autofac
>     Assembly Version: 4.9.4.0
>     Win32 Version: 1.11
>     CodeBase: file:///C:/Users/WINUSER/pass-winmenu/pass-winmenu.exe
>  
> System.IO.Abstractions
>     Assembly Version: 10.0.0.0
>     Win32 Version: 1.11
>     CodeBase: file:///C:/Users/WINUSER/pass-winmenu/pass-winmenu.exe
>  
> McSherry.SemanticVersioning
>     Assembly Version: 1.3.0.0
>     Win32 Version: 1.11
>     CodeBase: file:///C:/Users/WINUSER/pass-winmenu/pass-winmenu.exe
>  
> System.Windows.Forms
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4270.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
>  
> System.Drawing
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4084.0 built by: NET48REL1
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
>  
> YamlDotNet
>     Assembly Version: 8.0.0.0
>     Win32 Version: 1.11
>     CodeBase: file:///C:/Users/WINUSER/pass-winmenu/pass-winmenu.exe
>  
> Anonymously Hosted DynamicMethods Assembly
>     Assembly Version: 0.0.0.0
>     Win32 Version: 4.8.4250.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/mscorlib/v4.0_4.0.0.0__b77a5c561934e089/mscorlib.dll
>  
> LibGit2Sharp
>     Assembly Version: 0.26.0.0
>     Win32 Version: 1.11
>     CodeBase: file:///C:/Users/WINUSER/pass-winmenu/pass-winmenu.exe
>  
> Newtonsoft.Json
>     Assembly Version: 12.0.0.0
>     Win32 Version: 1.11
>     CodeBase: file:///C:/Users/WINUSER/pass-winmenu/pass-winmenu.exe
>  
> UIAutomationTypes
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4270.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/UIAutomationTypes/v4.0_4.0.0.0__31bf3856ad364e35/UIAutomationTypes.dll
>  
> UIAutomationProvider
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4270.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/UIAutomationProvider/v4.0_4.0.0.0__31bf3856ad364e35/UIAutomationProvider.dll
>  
> PresentationFramework-SystemXml
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4084.0
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework-SystemXml/v4.0_4.0.0.0__b77a5c561934e089/PresentationFramework-SystemXml.dll
>  

When I click _Push to Remote_ the push will fail and I get this error:

> ************** Exception Text **************
> System.NullReferenceException: Object reference not set to an instance of an object.
>    at PassWinmenu.ExternalPrograms.Git.CommitFile(String filePath)
>    at PassWinmenu.ExternalPrograms.Git.Commit()
>    at PassWinmenu.Actions.CommitChangesAction.Execute()
>    at System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
>    at System.Windows.Forms.ToolStripMenuItem.OnClick(EventArgs e)
>    at System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
>    at System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
>    at System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
>    at System.Windows.Forms.ToolStripDropDown.OnMouseUp(MouseEventArgs mea)
>    at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
>    at System.Windows.Forms.Control.WndProc(Message& m)
>    at System.Windows.Forms.ToolStrip.WndProc(Message& m)
>    at System.Windows.Forms.ToolStripDropDown.WndProc(Message& m)
>    at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
> 
> 
> ************** Loaded Assemblies **************
> mscorlib
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4250.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
>  
> pass-winmenu
>     Assembly Version: 1.11.0.0
>     Win32 Version: 1.11
>     CodeBase: file:///C:/Users/WINUSER/pass-winmenu/pass-winmenu.exe
>  
> PresentationFramework
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4270.0
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework/v4.0_4.0.0.0__31bf3856ad364e35/PresentationFramework.dll
>  
> WindowsBase
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4270.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/WindowsBase/v4.0_4.0.0.0__31bf3856ad364e35/WindowsBase.dll
>  
> System.Core
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4270.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
>  
> System
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4200.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
>  
> PresentationCore
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4270.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/PresentationCore/v4.0_4.0.0.0__31bf3856ad364e35/PresentationCore.dll
>  
> System.Xaml
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4270.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xaml/v4.0_4.0.0.0__b77a5c561934e089/System.Xaml.dll
>  
> System.Configuration
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4190.0 built by: NET48REL1LAST_B
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
>  
> System.Xml
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4084.0 built by: NET48REL1
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
>  
> PresentationFramework.Aero2
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4084.0 built by: NET48REL1
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework.Aero2/v4.0_4.0.0.0__31bf3856ad364e35/PresentationFramework.Aero2.dll
>  
> Autofac
>     Assembly Version: 4.9.4.0
>     Win32 Version: 1.11
>     CodeBase: file:///C:/Users/WINUSER/pass-winmenu/pass-winmenu.exe
>  
> System.IO.Abstractions
>     Assembly Version: 10.0.0.0
>     Win32 Version: 1.11
>     CodeBase: file:///C:/Users/WINUSER/pass-winmenu/pass-winmenu.exe
>  
> McSherry.SemanticVersioning
>     Assembly Version: 1.3.0.0
>     Win32 Version: 1.11
>     CodeBase: file:///C:/Users/WINUSER/pass-winmenu/pass-winmenu.exe
>  
> System.Windows.Forms
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4270.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
>  
> System.Drawing
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4084.0 built by: NET48REL1
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
>  
> YamlDotNet
>     Assembly Version: 8.0.0.0
>     Win32 Version: 1.11
>     CodeBase: file:///C:/Users/WINUSER/pass-winmenu/pass-winmenu.exe
>  
> Anonymously Hosted DynamicMethods Assembly
>     Assembly Version: 0.0.0.0
>     Win32 Version: 4.8.4250.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/mscorlib/v4.0_4.0.0.0__b77a5c561934e089/mscorlib.dll
>  
> LibGit2Sharp
>     Assembly Version: 0.26.0.0
>     Win32 Version: 1.11
>     CodeBase: file:///C:/Users/WINUSER/pass-winmenu/pass-winmenu.exe
>  
> Newtonsoft.Json
>     Assembly Version: 12.0.0.0
>     Win32 Version: 1.11
>     CodeBase: file:///C:/Users/WINUSER/pass-winmenu/pass-winmenu.exe
>  
> UIAutomationTypes
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4270.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/UIAutomationTypes/v4.0_4.0.0.0__31bf3856ad364e35/UIAutomationTypes.dll
>  
> UIAutomationProvider
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4270.0 built by: NET48REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/UIAutomationProvider/v4.0_4.0.0.0__31bf3856ad364e35/UIAutomationProvider.dll
>  
> PresentationFramework-SystemXml
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.4084.0
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/PresentationFramework-SystemXml/v4.0_4.0.0.0__b77a5c561934e089/PresentationFramework-SystemXml.dll
>  

Also when I change a password through the _Edit Password File_ option it will show me a smaller error when I click _save_, saying only: 

> Unable to save your password (encryption failed): Object reference not set to an instance of an object.

Now I just use the _Open Shell_ button to manually update my external repo with `git`, which works fine. When I look at `git status` it seems that some passwords do not get staged for commit. Maybe this is related to the problem.

I tried looking in past issues to see if this problem has already been mentioned before but I was not sure. If it is already a known problem then I apoligise.

[edit] fixed annoying text styling

**Comment by geluk** [comment]
Thanks for your report! I think the passwords not getting staged is indeed related to this issue. I'm going to investigate this a bit more.

**Comment by geluk** [comment]
I haven't been able to discover the underlying cause yet, but I created a new build which should at least prevent the application from crashing if the same condition occurs. Instead, it will report an error in this case.
You can download the new version here: https://ci.appveyor.com/api/buildjobs/rar2agcxh6jc8y1v/artifacts/pass-winmenu%2Fbin%2FRelease%2Fpass-winmenu.exe

Could you please test it again with this build? If you get an error again, please report the error message and the logs (right click the notification area icon -> `More actions` -> `View logs`). Thanks!

**Comment by phwoomp** [comment]
Hey geluk,

I tested again with the provided build like you asked . Now everything works perfectly! I get no errors whatsoever and the passwords are being pushed and pulled just like they should. 

Let me know if you need more info to find the underlying cause.

Thanks

**Comment by geluk** [comment]
That's good to hear! I did implement a fix for what I thought was a related issue, which is why I thought yours would probably still be present. However, that clearly ins't the case, otherwise you would have seen an error this time.
I'll round up these recent fixes and release a new version. Until then, the easiest solution is probably to continue using this build for now.

**Comment by FranekStark** [comment]
Hey guys,
I receive the same error (unfortunately it's in german):
```
Informationen über das Aufrufen von JIT-Debuggen
anstelle dieses Dialogfelds finden Sie am Ende dieser Meldung.

************** Ausnahmetext **************
System.NullReferenceException: Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt.
   bei PassWinmenu.ExternalPrograms.Git.CommitFile(String filePath)
   bei PassWinmenu.Windows.DialogCreator.AddPassword()
   bei PassWinmenu.WinApi.Notifications.<>c__DisplayClass11_0.<AddMenuActions>b__1(Object sender, EventArgs args)
   bei System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
   bei System.Windows.Forms.ToolStripMenuItem.OnClick(EventArgs e)
   bei System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   bei System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
   bei System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
   bei System.Windows.Forms.ToolStripDropDown.OnMouseUp(MouseEventArgs mea)
   bei System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   bei System.Windows.Forms.Control.WndProc(Message& m)
   bei System.Windows.Forms.ToolStrip.WndProc(Message& m)
   bei System.Windows.Forms.ToolStripDropDown.WndProc(Message& m)
   bei System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

```

**Comment by geluk** [comment]
@FranekStark I think you have the same issue, can you try the build I linked above to see if it fixes it for you as well?

**Comment by FranekStark** [comment]
@geluk yes, now it's also working for me!

**Comment by geluk** [comment]
I've just published [v1.11.1](https://github.com/geluk/pass-winmenu/releases/tag/v1.11.1), which fixes this issue.

I'll close the issue now, but if the problem persists, feel free to open a new issue, or to reopen this one.
